### ISSUE TITLE
Applicable policy / procedure for additional classes and associations behind https://www.opengis.net/def/schema/?

### ISSUE THREAD

**Issue by dblodgett-usgs** [opengeospatial/NamingAuthority#138]
The HY_Features SWG and other environmental domain feature model SWGs need to publish definitions of classes and associations for use in structured data. This is a well documented need that has been in "experimental" status for years. At a minimum, a URL that resolves to a text document containing a name is required and would be sufficient for initial use.

The content currently published here: https://www.opengis.net/def/schema/hy_features/hyf is in status "experimental" and -- as far as I know, there is no policy for governance of that or additional content there.

Am I missing a policy for publishing additional definitions? 

How should we go about establishing such a policy / procedure?

**Comment by dblodgett-usgs** [comment]
@ghobona -- do you have any response here? We have an email thread going with the HY_Features SWG to discuss whether the defs server is the appropriate place to publish URIs for the concepts of HY_Features and all and could use some input.

**Comment by ghobona** [comment]
@dblodgett-usgs The 'experimental' label meant that the definitions were the subject of an on-going Innovation initiative. Now that the initiative(s) have ended, please ask the relevant DWG or SWG to pass a motion approving the definitions for registration. OGC-NA will then mirror the vote and change the status of the definitions to 'valid'. 

Since we are a couple of months away from the next OGC Members' Meeting, I would suggest asking the Chair of the DWG or SWG to run an email vote. I will mirror the DWG/SWG vote with an OGC-NA vote once the DWG/SWG has approved the definitions. We can complete the whole process within a month.

**Comment by lieberjosh** [comment]
There seems to be a policy issue here. The HYF conceptual model (and UML) are an adopted standard. Any OWL or other implementation of the conceptual model is, well, an unapproved implementation. Is there an automated procedure sufficient to uniquely define an encoding such as OWL based on adopted UML, or should such an encoding still go through an adoption process? There has certainly been discussion of the need for judgement in creating ontologies from UML, even if it doesn't involve filling out a logical / physical model with properties, etc.

**Comment by ghobona** [comment]
@dblodgett-usgs My recommendation would be to register what is already there, even if it is half of the HY_Features model. The SWG (not the OGC-NA) would then incrementally build the rest of the ontology over a period of time. 

@lieberjosh OGC-NA policies allow for ontologies and definitions of concepts to be registered on the Definitions Server. Neither the ontologies nor the definitions have to be adopted standards prior to registration.

**Comment by dblodgett-usgs** [comment]
Thanks @ghobona. 

So then additional definitions would be contributed by modifying all the artifacts here: https://github.com/opengeospatial/NamingAuthority/tree/master/definitions/schema/hy_features/hyf ? 

This html, for example, looks like it's auto generated by something? https://github.com/opengeospatial/NamingAuthority/blob/master/definitions/schema/hy_features/hyf/hyf.html

**Comment by ghobona** [comment]
@dblodgett-usgs Yes, additional definitions would be contributed by modifying the artifacts [there](https://github.com/opengeospatial/NamingAuthority/tree/master/definitions/schema/hy_features/hyf). 

I can create a branch in this repository for the SWG to work from. Then when the SWG is ready to do a major update, the SWG can create a Pull Request and endorse the Pull Request through a SWG Motion. I or @rob-metalinkage can then review the Pull Request and recommend approval by the OGC-NA.

**Comment by dblodgett-usgs** [comment]
@rob-metalinkage -- do you mind removing the artifacts that are not required from this directory? https://github.com/opengeospatial/NamingAuthority/tree/master/definitions/schema/hy_features/hyf

**Comment by KathiSchleidt** [comment]
@ghobona picking up a relevant comment by @ingosimonis in the related mail threads:

`What we have not really worked out is the process for Definition Server updates.`

Unfortunately, from my experience, I can only confirm this! While I'm most thankful to @rob-metalinkage for reaching out and helping me navigate the def-server jungle, nice would be a clear process for provision and update of def server concepts, not requiring support from "native guides" ;) 

Any chance of somehow splitting the more experimental work that Rob is doing under the innovation program from the more pragmatic day-to-day concerns of getting resources published? I know one of my issues is seeing the forest for all the (experimental) trees in the way! I'd much appreciate a clean page somewhere detailing:

1. what concepts can be provided via the OGC def server
2. how these concepts are to be provide for ingestion (following which ttl, provided where?)
3. a simple but complete example for guidance in creating the necessary ingestion resources
 
I appreciate the lack of resources at OGC for such work, but I'd also like some appreciation for all the work SWG members are doing (on their own resources) trying (and in many cases failing) to get essential content provided (I know I've lost weeks on this to date, still failing :( )

Any outlook on remediation of this core OGC issue? We'll also be investigating options via the newly instated Conceptual Modelling Group running under the Architecture DWG

:)

Kathi

PS: something is also going strangely sideways in the [official def server content](http://defs.opengis.net/vocprez/vocab/) - PipelineML types seem to have taken over the world, all concepts listed individually?

**Comment by dblodgett-usgs** [comment]
I don't think we are communicating clearly here. See the screen shot I put in my last comment. There are various html, rdf, jsonld, ttl, png, and css artifacts in that directory. 

What Kathi and I are asking is:
1. what is allowable in terms of kinds of concepts for which we could register definitions?
2. what technical artifact do we use to submit those definitions?
3. some pointer to how the technical artifacts that are required can be created.

I understand that it's not the OGC Staff's role to decide what will be supported -- but as the designers and implementers of the register, you do possess knowledge that needs to be communicated if the community is going to engage with the register at all.

**Comment by ghobona** [comment]
_what is allowable in terms of kinds of concepts for which we could register definitions according to current policy?_

Section 5.4 of the [OGC-NA Charter](https://portal.ogc.org/files/?artifact_id=77804&version=1) lists the resources that can be registered by the OGC-NA.

_what technical artifact do we use to submit those definitions?_

Section 7.2 of the [OGC-NA Procedures policy](https://docs.opengeospatial.org/pol/09-046r6.html#_submission_process) describes how to submit proposals to the OGC-NA.

Once a proposal is received, I usually advise the SWG/DWG to create an Issue on the [GitHub board](https://github.com/opengeospatial/NamingAuthority/projects/2). Sometimes I create the Issue myself.

I then ask the SWG/DWG to place the artifact in the 'incubation' folder. Typically the artefacts are TTL but sometimes RDF/XML. Some of the proposals we received in the past were  CSV. Some of the proposals were GML Dictionaries. 

_some pointer to how the technical artifacts that are required can be created_

If all that is being proposed for registration is a series of codelists (or a flat register), then it is best to just submit a CSV file like [this one](https://github.com/opengeospatial/NamingAuthority/blob/master/incubation/SourceArchive/tileMatrixSets/tms.csv).

If what is being proposed for registration is a domain ontology, then it requires domain knowledge. The resources to create an ontology that is acceptable to the majority, even from an existing logical model, are enormous. That is why OGC Staff and the OGC-NA rely on the SWGs and DWGs to design the ontologies.
